methods.py

```diff
@@ -320,19 +320,31 @@ def split_lib(self, libname, src_list = None, env_lib = None):
     lib = env_lib.add_library(libname + str(num), list)
     lib_list.append(lib)
 
-    if len(lib_list) > 0:
-        if os.name == 'posix' and sys.platform == 'msys':
-            env.Replace(ARFLAGS=['rcsT'])
-            lib = env_lib.add_library(libname + "_collated", lib_list)
-            lib_list = [lib]
-
```

Comment:
Removing this was not strictly necessary for the use cases I tried (MinGW from MSYS2 on Windows and MinGW from Scoop in Powershell on Windows (both `nt` and `win32`), and MinGW from distro repos on Linux), but I'm not sure any recent MinGW distribution would be `posix` and `msys`.

So better drop this old code and make sure that all platforms that may use this method use the same code. If an issue pops up on a specific platform, we can handle it and document it as such.